src/main/java/org/apache/maven/plugins/dependency/ListClassesMojo.java

```diff
@@ -0,0 +1,326 @@
+
+        try
+        {
+            if ( transitive )
+            {
+                Iterable<ArtifactResult> artifacts = dependencyResolver
+                        .resolveDependencies( buildingRequest, coordinate, null );
+
+                for ( ArtifactResult result : artifacts )
+                {
+                    printClassesFromArtifactResult( result );
+                }
+            }
+            else
+            {
+                ArtifactResult result = artifactResolver
+                        .resolveArtifact( buildingRequest, toArtifactCoordinate( coordinate ) );
+
+                printClassesFromArtifactResult( result );
+            }
+        }
+        catch ( ArtifactResolverException | DependencyResolverException | IOException e )
+        {
+            throw new MojoExecutionException( "Couldn't download artifact: " + e.getMessage(), e );
+        }
+    }
+
+    private void printClassesFromArtifactResult( ArtifactResult result )
+            throws IOException
+    {
+        // open jar file in try-with-resources statement to guarantee the file closes after use regardless of errors
+        try ( JarFile jarFile = new JarFile( result.getArtifact().getFile() ) )
+        {
+            Enumeration entries = jarFile.entries();
+
+            while ( entries.hasMoreElements() )
+            {
+                JarEntry entry = (JarEntry) entries.nextElement();
+                String entryName = entry.getName();
+
+                // filter out files that do not end in .class
+                if ( !entryName.endsWith( ".class" ) )
+                {
+                    continue;
+                }
+
+                // remove .class from the end and change format to use periods instead of forward slashes
+                String className = entryName.substring( 0, entryName.length() - 6 ).replace( '/', '.' );
+                getLog().info( className );
+            }
+        }
+    }
+
+    private ProjectBuildingRequest makeBuildingRequest()
+            throws MojoExecutionException, MojoFailureException
+    {
+        if ( artifact == null )
+        {
+            throw new MojoFailureException( "You must specify an artifact, "
+                    + "e.g. -Dartifact=org.apache.maven.plugins:maven-downloader-plugin:1.0" );
+        }
+        if ( artifact != null )
+        {
+            String[] tokens = artifact.split( ":" );
+            if ( tokens.length < 3 || tokens.length > 5 )
+            {
+                throw new MojoFailureException( "Invalid artifact, you must specify "
+                        + "groupId:artifactId:version[:packaging[:classifier]] " + artifact );
+            }
+            coordinate.setGroupId( tokens[0] );
+            coordinate.setArtifactId( tokens[1] );
+            coordinate.setVersion( tokens[2] );
+            if ( tokens.length >= 4 )
+            {
+                coordinate.setType( tokens[3] );
+            }
+            if ( tokens.length == 5 )
```

Comment:
>= for safety

Reply:
Added via rebase of master. Should it be changed in this branch?